Dedupe the now-redundant CRT shaders' filename prefixes

kcgen opened the pull request:
With the shaders now organized by category into sub-directories, some of the filename prefixes are now redundant.

`glshader = crt/crt-easymode`

now collapses to:

`glshader = crt/easymode`

(Alternatively, if we wanted to keep the dupes, we should be consistent with category filename prefixes, i.e.: `glshader = scaler/scaler-advmame3x.glsl`)

---

Here's the tree:

```
glshaders/
├── crt
│   ├── aperture.glsl
│   ├── caligari.glsl
│   ├── easymode.glsl
│   ├── easymode.tweaked.glsl
│   ├── fakelottes.glsl
│   ├── fakelottes.tweaked.glsl
│   ├── geom.glsl
│   ├── geom.tweaked.glsl
│   ├── hyllian.glsl
│   ├── hyllian-updated.glsl
│   ├── lottes-fast.glsl
│   ├── lottes-fast.subtle+gain.glsl
│   ├── lottes.glsl
│   ├── lottes.tweaked.glsl
│   ├── mattias.glsl
│   ├── pi.glsl
│   ├── pi-vertical.glsl
│   ├── yee64.glsl
│   ├── yeetron.glsl
│   ├── zfast-composite.glsl
│   └── zfast.glsl
├── interpolation
│   ├── catmull-rom.glsl
│   └── sharp.glsl
├── scaler
│   ├── advinterp2x.glsl
│   ├── advinterp3x.glsl
│   ├── advmame2x.glsl
│   ├── advmame3x.glsl
│   ├── rgb2x.glsl
│   ├── rgb3x.glsl
│   ├── scan2x.glsl
│   ├── scan3x.glsl
│   ├── tv2x.glsl
│   └── tv3x.glsl
├── meson.build
└── none.glsl
```

johnnovak commented:
I agree with the consistency angle, this would be ideal. I started out by removing the prefixes but then added them back because I wanted to keep the exact same names from the upstream repo:

https://github.com/libretro/glsl-shaders/tree/master/crt

I guess alternatively we could still remove the prefixes and put an additional comment in the shader source files stating the original filenameto make backporting fixes easier.

johnnovak commented on src/dosbox.cpp:
```diff
@@ -645,10 +645,9 @@ void DOSBOX_Init()
-	        "directory. The '.glsl' extension can be omitted.");
+	        "Options include 'default', 'none', a shader listed using the --list-glshaders\n"
+	        "command-line argument, or an absolute or relative path to a file.\n"
+	        "In all cases, you may omit the shader's '.glsl' file extension.");
```
Nice! 👍🏻

kcgen commented:
The tie-breaker(s) for me (to consider this minor deduping) is that these filenames flow through into in our conf API, where users will be typing their names to select them.

Because they are public-facing, we are "putting them into action" within our project, as opposed to Tyrell's repo which was just a tree of files, without a project around them.

We care about consistent naming in our conf, as a priority - so another option would have been some kind of hardcoded name mapping (if we considered the source files as immutable; say like the FreeDOS code-pages).

But in this case, we have extra flexibility: Tyrells' repo is not maintained and your fixes haven't been merged - so ours become the latest pareto state of the files. 

> I guess alternatively we could still remove the prefixes and put additional comments in the shader source files stating the original filenames.

:+1:

Added. I updated all the CRT file headers to indicate that Tyrells ported them file from Libretro for DOSBox, and mentioned the origin GLSLP original filename.  I tried tailor each addition to suite the style of the original author (formatting nuances).

Hopefully it's OK now!

johnnovak commented:
> I updated all the CRT file headers to indicate that Tyrells ported them from Libretro for DOSBox, and mentioned the origin GLSLP original filename. I tried tailor each addition to suite the style of the original author (formatting nuances).

Sounds good, I think that's the best approach, and in retrospect I should've done it like this 👍🏻

johnnovak reviewed (approved):
Nice consistency & usability cleanup @kcgen 👍🏻

kcgen commented:
Thanks for the help with these, @johnnovak ! :rocket:
